app/Http/Controllers/Danabus/RouteController.php

```diff
@@ -23,9 +23,13 @@ public function index()
     /**
      * Display the specified resource.
      *
+     *  @param int $id of route
+     *
      * @return \Illuminate\Http\Response
      */
-    public function show()
+    public function show($id)
     {
+        $route = Route::where('id', '=', $id)->with('directions', 'directions.stop')->get();
```

Comment:
findOrFail is better
The detail page should use a single object not an array or collection